### PR TITLE
Use a retry loop to handle bucket build errors (STAGE-2315)

### PR DESCRIPTION
**Pull request by clupo**

### CODE REVIEW

**Comment by clupo** [comment]
Added a retry loop that tries 5 times and sleeps for 3 seconds between each try

**Comment by clupo** [comment]
I removed the timeout thing because it looked like the timeout didn't change, so we don't understand what the timeout is actually timeout out or that part of the libraries we're using is broken as well

And I made the change to create a test file and see if it was successfully created before continuing with building the actual data

**Comment by brantburnett** [comment]
@clupo 

I'd really prefer to put time timeout setting back.  Just because we don't need it doesn't mean that other users might not like it, and this is a public library/image.  And we've already released a version supporting that setting publicly, so removing it could cause somebody a problem.

**Comment by clupo** [comment]
I put the timeout back

**Comment by clupo** [comment]
No I didn't let me actually put it back

**Comment by clupo** [comment]
Okay you can look at it again

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,5 @@
-FROM couchbase:enterprise-5.0.0
+// prc 5.0.0 doesn't exist anymore
+FROM couchbase:enterprise-5.0.1
```
Comment (clupo) [inline Dockerfile]: prc 5.0.0 doesn't exist anymore

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM couchbase:enterprise-5.0.0
+FROM couchbase:enterprise-5.0.1
```
Comment (clupo) [inline Dockerfile]: 5.0.0 doesn't exist anymore

File: Dockerfile [hunk 3]
```diff
@@ -16,11 +16,15 @@ RUN wget https://github.com/stedolan/jq/releases/download/jq-1.5/jq-linux64 && \
 RUN npm install -g fakeit && \
     rm -rf /tmp/* /var/tmp/*
 	
+# Install nodejs couchbase package
+RUN npm install -g couchbase
```
Comment (clupo) [inline Dockerfile]: started getting this error `Cannot find module 'couchbase' require` at this call in the nodejs script `require('couchbase')`	so installing it through npm

File: Dockerfile [hunk 4]
```diff
@@ -16,11 +16,15 @@ RUN wget https://github.com/stedolan/jq/releases/download/jq-1.5/jq-linux64 && \
...
+	
 # Configure default environment
 ENV CB_DATARAM=512 CB_INDEXRAM=256 CB_SEARCHRAM=256 \
 	CB_SERVICES=kv,n1ql,index,fts CB_INDEXSTORAGE=forestdb \
 	CB_USERNAME=Administrator CB_PASSWORD=password \
-	FAKEIT_BUCKETTIMEOUT=5000
+	FAKEIT_BUCKETTIMEOUT=5000 \	
+	NODE_PATH=/usr/lib/node_modules
```
Comment (clupo) [inline Dockerfile]: Set this env so the couchbase package can be found

**Comment by clupo** [comment]
The 3 comments above are the 3 changes I made today

File: Dockerfile [hunk 5]
```diff
@@ -1,4 +1,4 @@
-FROM couchbase:enterprise-5.0.0
+FROM couchbase:enterprise-4.6.3
```
Comment (clupo) [inline Dockerfile]: 5.0.0 doesn't exist anymore

**Comment by clupo** [comment]
The 3 comments above are the 3 changes I made today for your review

**Pull request closed without merging**